src/server/index.ts

```diff
@@ -18,38 +18,49 @@ function analyzer(opts: AnalyzerPluginOptions = { analyzerMode: 'server' }): Plu
   const analyzerModule = createAnalyzerModule(opts?.gzipOptions)
   let defaultWd = process.cwd()
 
+  let previousSourcemapOption: any = false
```

Comment:
The value change in sync function, but it is used in async function. That might lead to async function get non-expected value.

Reply:
@sadofriod BTW `configResolved` in vite is a seq function. In us document we can know the plugin is work at the end. So i think is not a problem.